### DISCUSSION TITLE
Problems trying to get a working TLS connection

### DISCUSSION THREAD

**Discussion by SchubideiCroissant** [inspircd/inspircd#2113] [Support] [answered]
**Description**

 I am trying to setup my own IRC server with TLS on my local network.
The server runs on Ubuntu and is set to a static local address 192.168.2.214. Firewall is off (for now)
When i am trying to connect to the server via openssl s_client -connect, i get the following error 90470000:error:0A00010B:SSL routines:ssl3_get_record:wrong version number:../openssl-3.1.0/ssl/record/ssl3_record.c:354:.

**Steps to reproduce the issue:**

1.  Start the IRC Server
2. Connect via openssl s_client -connect or IRC service
3. Connection error

**Describe the results you received:**
So the problem probably lies in my inspircd.conf
It is located at:
Sydney@Raspi:~/inspircd-2.0.25/run/conf $ ls
cert.pem  dhparams.pem  examples  inspircd.conf  key.pem

Where all other crypt files are located as well.

**Describe the results you expected:**
Working connection.

**Additional information you deem important (e.g. issue happens only occasionally):**
I added the conf file in the appendix. (added a txt ending to upload it, which normaly isnt there)

**Output of `./bin/inspircd --version`:**

[inspircd.conf.txt](https://github.com/user-attachments/files/17096073/inspircd.conf.txt)

**Comment by sadiepowell** [comment]
It looks like you are trying to use v3+ SSL config on v2. If you upgrade to [a newer version](https://github.com/inspircd/inspircd/releases/latest) your SSL config should work.

**Reply by SchubideiCroissant** [reply, marked as answer]
Ty, idk how i got such an old version. Took a bit to take SSL working but it runs now :)
